### PR TITLE
Rework options, cli architecture, error management & more

### LINKED ISSUE #47: Rework binary cargo circus, move call to ocaml to driver

### PR DESCRIPTION
**Pull request by W95Psp**
- fixes #47
- related to #14

### CODE REVIEW

File: cli/src/cargo_circus.rs [hunk 1]
```diff
@@ -1,98 +1,67 @@
...
-use std::process::Command;
 
-const ENGINE_BINARY_NAME: &str = "circus-engine";
-const ENGINE_BINARY_NOT_FOUND: &str = const_format::formatcp!(
-    "The binary [{}] was not found in your [PATH].",
-    ENGINE_BINARY_NAME,
-);
+fn run(options: Options) -> i32 {
+    let exit_status = std::process::Command::new("cargo")
+        .args(["build".into()].iter().chain(options.cargo_flags.iter()))
```
Comment (franziskuskiefer) [inline cli/src/cargo_circus.rs:8]: Do you want to add the `+nightly-...` things later?

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, generally lgtm. A few comments and there are a few more things to do around running the driver with the right rust version etc. But we can move that to follow ups.

File: cli/src/cargo_circus.rs (hunk 1 above)
Comment (W95Psp) [reply cli/src/cargo_circus.rs:8]: Oh, maybe I can do a quick something already yes

Comment (W95Psp) [reply cli/src/cargo_circus.rs]: done in b433339ac576a4b89409aa38dea0d701476fcf23

File: cli/options/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+edition = "2021"
+build = "build.rs"
+
+[dependencies]
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
+schemars = "0.8"
+itertools = "0.10.5"
+clap = { version = "4.0", features = ["derive"] }
+# circus-frontend-exporter = { path = "../../frontend/exporter" }
```
Comment (franziskuskiefer) [inline cli/options/Cargo.toml:13]:
```suggestion
```

**Review by franziskuskiefer — APPROVED** [review]
Thanks! Lets get this in.

**Pull request merged**